zproject/prod_settings_template.py

```diff
@@ -211,6 +212,139 @@
+#     bVugryTGIwKBgGMpukMEEELDvylDu/j9Oo9a51iiXIT9PpcN2tZjFGUzFiU+JRNM
+#     O791d3ktXKMghqb7ehzbDYzOoqotGz28VFAYwBQTxsHyv3c5LY/iOKdajMirw33k
+#     Vn1o2FtRM1bh1goTfYF9ZpRir+P2N/j25LdlN4Fmc1ENWVoTXQQEa1VhAoGBAMqr
+#     nF/ctMQg7FkQZAsu9OkceMw9JPxYP+9SnBgY4kPVg33jZ9XHiQ1u1FnJS7E1RgQX
+#     nq+PHtGX/hepIM8GBIwsr3oPFiwJBuIX3oduM92mobbTU4GAGCrMLafSArk2ZcQw
+#     y0xP9VuHxvrOVKO+bILZGcslZ74IvvFBderW3hyjAoGBAMEkcZLfe2+FmAe2fju0
+#     F1qCAWqnqpVAgWsuDycS6k5TYHj/6vm8XVx+Xa8r/IfTuyXlAq/CPb4KPPY4ZAUy
+#     iAHVqzslbBuBOHaukySepy2Z4GBaMoD742H0QBdaNQE8pBjZhyir+paTbkehd5kT
+#     8LJ2ctg7ipCEcGe5b+gEllTe
+#     """.replace(' ', '').replace('\n', '').strip()
+
+# SOCIAL_AUTH_SAML_ORG_INFO =  {
+#     "en-US": {
+#         "name": "example",
+#         "displayname": "Example Inc.",
+#         "url": "%s%s" % ('https://', EXTERNAL_HOST),
+#     }
+# }
+
+# SOCIAL_AUTH_SAML_TECHNICAL_CONTACT = {
+#     "givenName": "Tech Gal",
+#     "emailAddress": "technical@example.com"
+# }
+
+# SOCIAL_AUTH_SAML_SUPPORT_CONTACT = {
+#     "givenName": "Support Guy",
+#     "emailAddress": "support@example.com",
+# }
+
+# Important note: Only one IdP is currently supported, don't add
+# any extra IdPs in this setting.
+# SOCIAL_AUTH_SAML_ENABLED_IDPS = {
+#     "testshib": {
+#         "entity_id": "https://idp.testshib.org/idp/shibboleth",
+#         "url": "https://idp.testshib.org/idp/profile/SAML2/Redirect/SSO",
+#         "x509cert": \
+#             """
+#             MIIDojCCAoqgAwIBAgIGAW0svbVqMA0GCSqGSIb3DQEBCwUAMIGRMQswCQYDVQQGEwJVUzETMBEG
+#             A1UECAwKQ2FsaWZvcm5pYTEWMBQGA1UEBwwNU2FuIEZyYW5jaXNjbzENMAsGA1UECgwET2t0YTEU
+#             MBIGA1UECwwLU1NPUHJvdmlkZXIxEjAQBgNVBAMMCXp1bGlwY2hhdDEcMBoGCSqGSIb3DQEJARYN
+#             aW5mb0Bva3RhLmNvbTAeFw0xOTA5MTMyGWRTffNaFw0yOTA5MTMyMjI4MTNaMIGRMQswCQYDVQQG
+#             EwJVUzETMBEGA1UECAwKQ2FsaWZvcm5pYTEWMBQGA1UEBwwNU2FuIEZyYW5jaXNjbzENMAsGA1UE
+#             CgwET2t0YTEUMBIGA1UECwwLU1NPUHJvdmlkZXIxEjAQBgNVBAMMCXp1bGlwY2hhdDEcMBoGCSqG
+#             SIb3DQEJARYNaW5mb0Bva3RhLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAL3K
+#             E57ZDDVplrZO1RKpz9zFekhXZZFiPhW2TCTtoaI966sGaRCmV10cb1FCUxxI3ilcjY8G5irHYc5O
+#             D4S8+FeIaHb036VjtZZNCDkamE2zGZCix5wCpXhxhQrXkkPJbzO49hhDF3HHJJPwefGfYnPC8/Oj
+#             bZ0OUuR8KkNbgn2VnqwZtmb0EX5xrA+212UDyVQ7izVXOoBbvzeydLh8EWteEXjKBREKGBfCL9Kl
+#             x8JY7BlYrZx+13NeDQsL7bgTXMnTIp3MVP3xddRqsatwersRVGr9b/HzXxfwu/MU230swjsNlgLZ
+#             OXiYD43rNEkJRlFfMnlY8F3IoE1Mki2BJtMCAwEAATANBgkqhkiG9w0BAQsFAAOCAQEAI8Xk13NT
+#             rjo417U1+wZjUKqvB7iw4zGapMqpGRVavGw9hZmSCs8/AJAkFZQKMhAR9GGqf7JHHj/4fNEQ+XVh
+#             YF1jCUR/X2VwUiBseDHaUKj7EZiX9tIFEI/6LVfPRjKNy1RkEXHo7Lg4RnctclZ1KU7mIZkPSk1J
+#             fShKIUhtvNaCYJ4OVkN+giQQ6u9HwBqoBYikOBhvgXfIlBFD5H1n7JqxOjWZNO7Rhhx+TjD/0Dmd
+#             BE04J2bv5zCllBWSv2e1YFi+5SBTBq6FaIMz5c8T4WRFpRlnDEvvEREeAsAbaDiYvpJlO6hOzHNj
+#             KVmHpmQsDhsgXP02tDsfTARf3EXhbA==
```

Comment:
I think we probably want the docs to instruct the user to put the certificate at a path, and have the code here read from that path.  E.g. tell them to put the cert at `/etc/zulip/saml-cert.crt` and then have this block contain e.g.:
```
if os.path.exists(SAML_CERT_PATH):
    with open(SAML_CERT_PATH, "r") as f:
        SAML_CERT = f.read()

...
`"x509cert": SAML_CERT,
```

(Or maybe more complex names since it looks like there's more than one).

Reply:
I added a commit (not properly tested yet, but it shows the approach) changing things to this sort of scheme. What I went for is putting the server certificate and private key in ``/etc/zulip/saml/`` and IdP certificates in ``/etc/zulip/saml/idps/idp_name.crt``.

Reply:
Sounds reasonable.